learningequality/studio
# Handle updates from IndexedDB in Vuex

## Summary
### Description of the change(s) you made

- Adds Vuex mutations that handle Dexie change events coming from IndexedDB updates
- Adds insurance to include `contentnode` table when applying changes
- Fixes issues with saving and overwriting of `learnerManaged` and completion criteria in the edit modal


### Manual verification steps performed
1. For each affected Vuex plugin, make associated changes that would trigger updates to the IndexedDB
2. Ensure that data is not overwritten

### Screenshots (if applicable)


### Does this introduce any tech-debt items?

___
## Reviewer guidance
### How can a reviewer test these changes?



### Are there any risky areas that deserve extra testing?

- Making edits to all the new metadata
- Making edits to completion criteria
- Making edits to a channel's content defaults

## References

Fixes https://github.com/learningequality/studio/issues/3511

## Comments

There appears to be an issue with categories, that I avoided because of it being a potential rabbit hole. When you select a top level category, then a child, the child category doesn't always immediately become selected. Clicking it again selects it, but this could cause some user aggravation.

----

### Contributor's Checklist


PR process:


Studio-specifc:



Testing:

___

### Reviewer's Checklist
#### This section is for reviewers to fill out.
